### ISSUE TITLE
Unable to provision PrivateDnsZoneGroup

### ISSUE THREAD

**Issue by dekobon** [pulumi/pulumi-azure-nextgen#227]
When attempting to provision a private DNS zone group, I get the following error:
```
Diagnostics:
  azure-nextgen:network/latest:PrivateDnsZoneGroup (z-ncdb):
    error: cannot create already existing resource '/subscriptions/<redacted>/resourceGroups/rg-nc-elijahz22/providers/Microsoft.Network/privateEndpoints/ep-nctodb-elijahz22/privateDnsZoneGroups/z-ncdb-elijahz22'
 
  pulumi:pulumi:Stack (<redacted>):
    Azure Location: WestUS
    error: update failed

```

Interestingly, when I use resources.azure.com to try to navigate to the resource name shown in the error message, I found that there was no resource by that name. In fact, there is no value for `privateEndpoints` under the `../Microsoft.Network/` provider.

Previously, this all worked in the Terraform based provider. I would do the following to assemble a private link from my VM to a PostgreSQL database service.
```python
    private_dns_zone = privatedns.Zone(
        name='privatelink.postgres.database.azure.com',
        resource_name='z-ncdb',
        resource_group_name=resource_group.name)
    private_dns_vnet_link = privatedns.ZoneVirtualNetworkLink(
        resource_name='zvnl-ncdb',
        resource_group_name=resource_group.name,
        virtual_network_id=net.id,
        private_dns_zone_name=private_dns_zone.name,
        registration_enabled=True,
    )
    db_private_link = privatelink.Endpoint(
        resource_name='ep-nctodb',
        resource_group_name=resource_group.name,
        subnet_id=subnet.id,
        private_service_connection=privatelink.EndpointPrivateServiceConnectionArgs(
            name='psc-nctodb',
            private_connection_resource_id=db_server.id,
            subresource_names=['postgresqlServer'],
            is_manual_connection=False),
        private_dns_zone_group=privatelink.EndpointPrivateDnsZoneGroupArgs(
            name='pdnsg-ncdb',
            private_dns_zone_ids=[private_dns_zone.id]),
        opts=ResourceOptions(depends_on=[subnet, private_dns_zone, private_dns_vnet_link]))
```

This is what I'm doing in the azure-nextgen provider:
```python
private_endpoint_resource = network.PrivateEndpoint(
    resource_name='ep-nctodb',
    resource_group_name=resource_group.name,
    private_endpoint_name='ep-nctodb-{0}'.format(installation_id),
    location=location,
    private_link_service_connections=[network.PrivateLinkServiceConnectionArgs(
        group_ids=['postgresqlServer'],
        private_link_service_id=db_server.id,
        name="psc-nctodb",
        private_link_service_connection_state={
            'actions_required': 'None',
            'description': 'Auto-approved',
            'status': 'Approved',
        },
    )],
    subnet=network.SubnetArgs(id=subnet.id)
)
private_dns_zone = network.PrivateDnsZoneGroup(
    resource_name='z-ncdb',
    resource_group_name=resource_group.name,
    private_dns_zone_group_name='z-ncdb-0'.format(installation_id),
    private_dns_zone_configs=[network.PrivateDnsZoneConfigArgs(
        name='privatelink.postgres.database.azure.com'
    )],
    private_endpoint_name=private_endpoint_resource.name
)
private_dns_vnet_link = network.VirtualNetworkLink(
    resource_name='zvnl-ncdb',
    resource_group_name=resource_group.name,
    location='global',
    virtual_network_link_name='{0}/zvnl-ncdb-'.format('privatelink.postgres.database.azure.com', installation_id),
    virtual_network=network.SubResourceArgs(
        id=net.id),
    private_zone_name=private_dns_zone.name,
    registration_enabled=True,
)
```

There may be an element of user error here, but I do think at the least there is a usability problem because it is difficult to figure out how to properly compose this pattern.

**Comment by mikhailshilkov** [comment]
Thank you for reporting this @dekobon 

I can reproduce the issue. The problem is that the Azure service returns an HTTP response 200 for non-existing DNS Zone Groups, with an empty body. Our current check expects a 404 in that case, so it does not proceed. I think we'll have to change that to parsing the 200 response and validating it has a reasonable shape. That's what the Terraform provider does in this case.

I'll prioritize the fix for this.

**Comment by dekobon** [comment]
For what it is worth, I'm happy to beta test any fixes.

**Comment by mikhailshilkov** [comment]
Fixed in the upcoming 0.5.0

**Comment by mikhailshilkov** [comment]
> For what it is worth, I'm happy to beta test any fixes.

I think it should work if you try this pre-release version https://pypi.org/project/pulumi-azure-nextgen/0.4.1a1611308823/
